slsDetectorServers/slsDetectorServer/src/nios.c

```diff
@@ -126,4 +129,50 @@ int mapCSP0(void) {
     return OK;
 }
 
-u_int32_t *Nios_getBaseAddress() { return csp0base; }
\ No newline at end of file
+u_int32_t *Nios_getBaseAddress() { return csp0base; }
+
+int Nios_checkKernelVersion(char *expectedVersion) {
+    // extract kernel date string
+    struct utsname buf;
+    if (uname(&buf) == -1) {
+        LOG(logERROR, ("Could not get kernel version\n"));
+        return FAIL;
+    }
+
+    // remove first word (#version number)
+    const char *ptr = strchr(buf.version, ' ');
+    if (ptr == NULL) {
+        LOG(logERROR, ("Could not parse kernel version\n"));
+        return FAIL;
+    }
+    char output[256];
+    memset(output, 0, 256);
```

Comment:
Not that this is a performance critical part of the code, but could we pass just the pointer into the buf.version string instead of making a copy here

On the other hand since we access it multiple times why not.